dask_kubernetes/core.py

```diff
@@ -275,11 +280,37 @@ def asynchronous(self):
         )
 
         if self.n_workers:
-            await self._scale(self.n_workers)
+            # will call KubeCluster.scale_up/scale_down
+            self.scale(n_workers)
+
+    async def periodic(self):
+        """Periodic Callback to check cluster has the correct size
+        """
+        # peridically check cluster is at the correct size
+        # don't assign member variables or collisions may happen
+        while True:
+            pods = await self._pods()
+            logger.debug(f"Current Cluster Size: {len(pods)} Cluster Target: {self._manual_scale_target}")
+            if self.task_queue.empty():
+                if (len(pods) != self._manual_scale_target):
+                    self.scale(self._manual_scale_target)
+            else:
+                # Get a scale_up/down out of the queue.
+                func = await self.task_queue.get()
+                # execute function
+                await func()
+                # cleanup messy pods
+                pods = await self._cleanup_terminated_pods(pods)
+
+                self.task_queue.task_done()
+
+            await asyncio.sleep(1)
```

Comment:
I wonder if we might wait here for either 1s or when there is a new element in the task queue, whichever is first.  That might help to improve resonsiveness.